### PR TITLE
Add support for ADC using voucher_client

### PR DESCRIPTION
**Pull request by rxbchen**
CloudBuild has some issues retrieving id_tokens from the metadata server resulting in the following error:

```
creating client failed: metadata: GCE metadata "instance/service-accounts/default/identity?audience=[someurl]" not defined
```

This pr aims to make auth using application-default-credentials possible, since it is exposed by GCB

### CODE REVIEW

File: v2/client/client.go [hunk 1]
```diff
@@ -42,6 +42,12 @@ func NewAuthClient(voucherURL string) (*Client, error) {
 	return newClient(voucherURL, authClient)
 }
 
+// NewCustomClient creates a new Client set to connect to passed
+// hostname using a passed http client
+func NewCustomClient(voucherURL string, client *http.Client) (*Client, error) {
```
Comment (thepwagner) [inline v2/client/client.go:47]: Idea: In https://github.com/grafeas/voucher/pull/52 , I was pondering with introducing [functional options](https://golang.cafe/blog/golang-functional-options-pattern.html) as means of wrangling the options for building a client.

I talked myself out of it there, but a third constructor is a sign it's worth considering? I'm a fan of "rule of threes" refactoring.

File: v2/cmd/voucher_client/defaultclient.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		return nil, err
+	}
+	ts := oauth2.ReuseTokenSource(nil, &idTokenSource{TokenSource: src})
+	return ts, nil
+}
+
+func newHttpClient() (*http.Client, error) {
+	ts, err := getDefaultTokenSource()
+	if err != nil {
+		return nil, fmt.Errorf("error creating client: %w", err)
```
Comment (thepwagner) [inline v2/cmd/voucher_client/defaultclient.go]:
```suggestion
		return nil, fmt.Errorf("error creating token source: %w", err)
```

Nit: differentiate between this error case and :65.

**Review by thepwagner — APPROVED** [review]
LGTM. I _think_ this will also respect `GOOGLE_APPLICATION_CREDENTIALS` envvar, and I dig that.

Are you planning to send this to `grafeas/voucher`? Seems worth it to me!

This has some conflicts with https://github.com/grafeas/voucher/pull/52/ ; we should probably land this first, as that's just a nice-to-have. We could pair up on porting one+both of those to functional options, or :shipit: with multiple constructors.

**Comment by rxbchen** [comment]
I'm doing this here so I can build + test it first with cloudbuild! Don't want to find out down the line that it doesn't work hahahah

File: v2/client/client.go (hunk 1 above)
Comment (rxbchen) [reply v2/client/client.go:47]: I like the idea of functional options (just read about it), and it seems like a really good approach!

**Comment by rxbchen** [comment]
superseded by: https://github.com/grafeas/voucher/pull/53

**Pull request closed without merging**